nuyer iP

nuyer opened the pull request:
#Dino Bot

### DinoPro
> “The first rule of management is delegation. Don’t try and do everything yourself because you can’t.” – Anthea Turner 

~Manage your own Tasks~
No worries ✋ 
Dino will keep track of all your _TODO tasks_! 🔥 ⭐ 
- CLI-friendly
- Easy to use
- Easy to learn


**Get Started now**:
- [x] download it from [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/413)
- [ ] double-click to launch
- [ ]  add your tasks
- [ ]  let Dino manage your tasks for you


**Features**:
1. Managing tasks
2. Managing deadlines (coming soon)
3. Reminders (coming soon)


For Java programmer, you can leverage DukePro as a tool to practice your skills. Here's the `main` method to get you started:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

nuyer committed:
Level 0. Rename, Greet, Exit

nuyer committed:
Level 1. Echo

nuyer committed:
Level 2. Add, List

nuyer committed:
Level 3. Mark as Done

nuyer committed:
Level 4. ToDos, Events, Deadlines

nuyer committed:
Level5. Handle Errors

nuyer committed:
A-TextUiTesting: Automated Text UI Testing

nuyer committed:
Level 6. Delete

nuyer committed:
restructure Duke.java and add enums

nuyer committed:
Implement Save to save and load tasks from cache

nuyer committed:
Change Enums to PascalCase

nuyer committed:
Correct indentation and coding style

sethteo commented:
Followed proper code conventions, no complicated expressions or nested conditional expressions :)))

nuyer committed:
Level 8. Dates and Times: store dates as a java.time.LocalDateTime

nuyer committed:
Create TaskList Class

nuyer committed:
Add storage class

nuyer committed:
Edit HORIZONTAL LINE constant

nuyer committed:
create tasks package and commands package

nuyer committed:
Create all command classes in command package

nuyer committed:
Merge commit 'a6f73244a20539e20c2189973ebf597df34ef631'

nuyer committed:
Setting up for JUnit: rearrange file

sethteo reviewed (commented):
Please refer to my most updated review :)

nuyer committed:
Restructure Command package for JUnit Test: change return type Command.execute to List<String>

nuyer committed:
Create JUnit test for Event and DeadlineCommand

nuyer committed:
Reorder Imports and Add getDescription and getisDone method

nuyer committed:
Reorder Imports and Add getDescription and getisDone method

nuyer committed:
Create FindCommand class

nuyer committed:
Edit Parser to Support Find Command

nuyer committed:
Edit FindCommand to show the order number

nuyer committed:
Add JavaDoc for classes and methods

nuyer committed:
Merge branch 'branch-A-CodingStandard' into branch-A-JavaDoc

YuSoonZ commented on src/main/java/duke/commands/Parser.java:
```diff
@@ -0,0 +1,65 @@
+            Parser.Instruction ins = toInstruction(parsedInput[0]);
+            String details = parsedInput.length > 1 ? parsedInput[1] : "";
+
+            switch (ins) {
```
Would need to adjust the switch block to align with this module's codestyle :)

YuSoonZ commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,67 @@
+package duke;
+
+import duke.tasks.TaskList;
```
Clean segregations of codes!

YuSoonZ reviewed (commented):
A well structured code that is easy to read and understand. Well done!

shamesjen commented on src/main/java/duke/commands/Parser.java:
```diff
@@ -0,0 +1,67 @@
+package duke.commands;
```
Should the parser be in its own package? Especially since it is not a command and does not extend Command

shamesjen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,67 @@
+package duke;
```
Should you rename the main file to match the name of your bot?

nuyer replied:
Thank you, I'll rename it!!

shamesjen commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,54 @@
+public class DeadlineCommand extends Command {
+    private static final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+    private final String details;
+    private final List<String> messages = new ArrayList<>();
```
I like how you use messages to compile all the messages together instead of using println for each individual message.

shamesjen commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,45 @@
+     * @param tasks A TaskList object that contains a list of tasks.
+     * @return List of Strings messages which will be printed to user later.
+     */
+    public List<String> execute(TaskList tasks) throws DukeException {
```
Should you add @ Override since you are overriding the abstract method from Command.java

shamesjen commented on src/main/java/duke/commands/Parser.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+
+    enum Instruction {
```
I like the way you used enum for your instructions

shamesjen commented on src/test/java/duke/DeadlineCommandTest.java:
```diff
@@ -0,0 +1,43 @@
+package duke;
```
Comprehensive and effective testing!

shamesjen reviewed (commented):
Great code that implements solid OOP principles. Do consider some of the suggestions I have made. 👍

nuyer committed:
Correct indentation

sethteo commented on src/main/java/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.format.DateTimeParseException;
+
+public class DeadlineCommand extends Command {
+    private static final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```
Good use of noun for variable used

sethteo commented on src/main/java/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,29 @@
+    DeleteCommand(String details) {
+        this.details = details;
+    }
+    public void execute(TaskList tasks) throws DukeException {
```
Perhaps it could be named executeTask() instead

nuyer replied:
I'll work on it. Thanks!

sethteo commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,23 @@
+
+public class Event extends Task implements Serializable {
+    private LocalDateTime start;
+    private LocalDateTime end;
```
Perhaps it could be better if it was startDate and endDate

sethteo reviewed (commented):
LGTM! Code follows the code quality guidelines closely. Most classes/variables are named as nouns while functions are verbs. Code was written with standard words. Little to no misleading names. Great job!

nuyer committed:
Correct Coding Standard by using Checkstyle
Reorder import in Duke.java

nuyer committed:
Add javafx

nuyer committed:
Setup basic JavaFX for duke

nuyer committed:
Styling the UI

nuyer committed:
Add user input event handlers

nuyer committed:
Add Dialog Boxes and Avatars

nuyer committed:
Add flip method in DialogBox such that the image on the left for Dino's output

nuyer committed:
Add padding in dialogbox annd clip avatar into a circle shape

nuyer committed:
Add background color to each dialog box

nuyer committed:
Adding Dialog Boxes

nuyer committed:
Merge Level-10 into master branch

nuyer committed:
Change filename and package name from duke to dino

nuyer committed:
Improve Code quality: Extract some methods

nuyer committed:
Change Duke to Dino and Refactor

nuyer committed:
Merge branch 'branch-fix-duke-to-dino'

nuyer committed:
Change correct duke to dino for the messages

nuyer committed:
Rename Image file

nuyer committed:
Merge branch 'branch-A-CodeQuality' into master

nuyer committed:
Add assertions to processInput() method

The function processInput is responsible for parsing user input and returning the corresponding command object based on the input. It is essential to ensure that the input and instruction are not null to prevent unexpected behavior.

Adding assertions for input and instruction checks enhances the robustness of the function and helps catch potential issues early.

* Ensure input string is not null or empty.
* Ensure Instruction enum is not null.

These assertions provide clarity and improve the reliability of the processInput function.

nuyer committed:
Merge branch 'branch-A-Assertions': Add assertions to enhance code reliability

The branch-A-Assertions introduces a series of assertions to the codebase, specifically within the processInput() methos. These assertions aim to improve the reliability  of the code.

Assert that the input string provided to the processInput function is not null or empty.
Assert that the Instruction enum obtained from the parsed input is not null.
These assertions serve as guardrails against unexpected behavior and programming errors, helping to catch potential issues early.

By merging branch-A-Assertions into the master branch, we integrate these enhancements into the main codebase, thereby improving its overall stability and maintainability.

nuyer committed:
Add Reminders class for task reminders

The Reminders class is introduced to provide reminders for tasks scheduled for today. It includes methods to filter out tasks from a TaskList, specifically Deadline and Event tasks, based on whether they are scheduled for today. The class also provides methods to display today's deadlines and events. This commit adds the Reminders class, which enhances the functionality of the Dino task manager by enabling users to receive reminders for tasks due today.

Changes:
* Added Reminders class to provide task reminders
* Implemented methods to filter out Deadline and Event tasks for today
* Included methods to display today's deadlines and events

nuyer committed:
Refactor method calls to utilize Reminder class for displaying today's events and deadlines

The getGreeting method in the specified class previously invoked methods from another class (Reminders) to fetch reminders associated with tasks. To improve code organization and adhere to better design principles, this commit refactors the method calls to utilize the Reminder class directly for fetching today's events and deadlines.

- Modify the Greeting method to invoke methods from the Reminder class.
- Specifically, call the methods for fetching today's events and deadlines.
- Update the method signature and parameters as necessary to accommodate the changes.

This refactoring enhances code clarity, encapsulation, and maintainability by consolidating related functionality within the Reminder class and promoting better separation of concerns.

nuyer committed:
Add User Guide

nuyer committed:
Add ByeCommand and improve implementation of handleUserInput

nuyer committed:
Improve Error messages

nuyer committed:
Improve Styling

nuyer committed:
Make the reminder message is displayed and sent out upon launching

nuyer committed:
Correct the output message for DeleteCommand

nuyer committed:
Add UI images

nuyer committed:
Add Javadoc for Reminder class

nuyer committed:
Update README.MD

nuyer committed:
Rename UI image files

nuyer committed:
Rename UI.PNG to Ui.PNG

nuyer committed:
Rename Ui.PNG to Ui.png

nuyer committed:
Integrate stream in getGreeting method

nuyer committed:
Merge pull request #2 from nuyer/A-Streams

A-Stream PR
